pkg/config/certificates.go

```diff
@@ -0,0 +1,151 @@
+package config
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"net/http"
+	"os"
+	"time"
+)
+
+type CertUser interface {
+	ClientCert() string
+	ClientKey() string
+	CACert() string
+	ClientCertPath() string
+	ClientKeyPath() string
+	CACertPath() string
+}
```

Comment:
I pushed this as a second commit because it over-complicate things a little, but I was trying to make using certs a little more streamlined in the future. Open to thoughts

Reply:
+1, i think this is a great idea!